app/screens/home/RouteList.tsx

```diff
@@ -185,6 +189,39 @@ const RouteList = ({
       .finally(() => setIsCryptoLoading((obj) => updateObject(obj, { [route.id]: false })));
   };
 
+  const fetchBuyHistory = (route: BuyRoute) => {
+    setRouteHistory(undefined);
+    setIsRouteHistoryVisible(true);
+    getBuyRouteHistory(route)
+      .then(setRouteHistory)
+      .catch(() => {
+        setIsRouteHistoryVisible(false);
+        NotificationService.error(t("feedback.load_failed"));
+      });
+  };
+
+  const fetchSellHistory = (route: SellRoute) => {
+    setRouteHistory(undefined);
+    setIsRouteHistoryVisible(true);
+    getSellRouteHistory(route)
+      .then(setRouteHistory)
+      .catch(() => {
+        setIsRouteHistoryVisible(false);
+        NotificationService.error(t("feedback.load_failed"));
+      });
+  };
+
+  const fetchCryptoHistory = (route: CryptoRoute) => {
+    setRouteHistory(undefined);
+    setIsRouteHistoryVisible(true);
+    getCryptoRouteHistory(route)
+      .then(setRouteHistory)
+      .catch(() => {
+        setIsRouteHistoryVisible(false);
+        NotificationService.error(t("feedback.load_failed"));
+      });
+  };
```

Comment:
A lot of duplicated code. Could/would you use a single method and fetch the correct history depending on the type?